processing/src/main/java/org/apache/druid/query/rowsandcols/column/DoubleArrayColumn.java

```diff
@@ -126,4 +74,122 @@ public <T> T as(Class<? extends T> clazz)
     }
     return null;
   }
+
+  private class MyColumnAccessor implements BinarySearchableAccessor
```

Comment:
Generically speaking, yes it would be best to validate that first before returning the thing.

As a confounding factor, even if the whole column isn't sorted, it could be sorted in the context of another column.  I.e. if a set of rows are sorted by `(col1, col2)`, then `col2` is not actually sorted if you look at the data, but when you access it in conjunction with `col1`, it is.  I agree that we should do more to figure this sort of thing out and fail if it's violated, but, for now, the code is counting on the query planning to do things correctly